components/url/origin.rs

```diff
@@ -101,7 +101,7 @@ pub struct OpaqueOrigin(Uuid);
 known_heap_size!(0, OpaqueOrigin);
 
 /// A representation of an [origin](https://html.spec.whatwg.org/multipage/#origin-2).
-#[derive(Clone, Debug)]
+#[derive(Clone, Debug, Serialize, Deserialize)]
```

Comment:
My compiler complains that `Serialize` and `Deserialize` are not implemented for `Host` when I attempt to derive for `MutableOrigin`. It seems like they are implemented tho: https://github.com/servo/rust-url/blob/master/src/host.rs#L93

I need this because `RequestInit` requires both `Serialize` and `Deserialize` from its types.